scripts/helpers/iwyu_check.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env -S python3 -B
+
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import glob
+import json
+import logging
+import os
+import re
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+import click
+import coloredlogs
+
+# Supported log levels, mapping string values required for argument
+# parsing into logging constants
+__LOG_LEVELS__ = {
+    'debug': logging.DEBUG,
+    'info': logging.INFO,
+    'warn': logging.WARN,
+    'fatal': logging.FATAL,
+}
+
+root_dir = os.path.dirname(os.path.realpath(__file__))
+proj_root_dir = os.path.join(Path(root_dir).parent.parent)
+
+
+def find_program(names):
+    for name in names:
+        found = shutil.which(name)
+        if found is not None:
+            return found
+
+
+@click.command()
+@click.option(
+    '--log-level',
+    default='INFO',
+    type=click.Choice(__LOG_LEVELS__.keys(), case_sensitive=False),
+    help='Determines the verbosity of script output.')
+@click.option(
+    '--no-log-timestamps',
+    default=False,
+    is_flag=True,
+    help='Skip timestaps in log output')
+@click.option(
+    '--compile-commands-glob',
+    default=os.path.join(proj_root_dir, "out", "debug", "compile_commands*.json"),
+    help='Set global pattern for compile_commands.json files'
```

Comment:
I would add `show_default=True` to the option arguments (where appropriate), so user will see default values as a guide for modifications. However, I'm not sure shince which version of click package the show_default is available.